instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/EvOfflineTest.kt

```diff
@@ -0,0 +1,153 @@
+            navigation.startTripSession()
+            withoutInternet {
+                val requestResult = navigation.requestRoutes(evRouteInBerlin())
+                    .getSuccessfulResultOrThrowException()
+                assertEquals(RouterOrigin.Onboard, requestResult.routerOrigin)
+                navigation.setNavigationRoutesAsync(requestResult.routes)
+            }
+            stayOnPosition(
+                longitude = 13.361378213031003,
+                latitude = 52.49813341962201
+            ) {
+                val onlineAlternative = navigation.alternativesUpdates()
+                    .filterIsInstance<NavigationRouteAlternativesResult.OnRouteAlternatives>()
+                    .filter { it.routerOrigin == RouterOrigin.Offboard }
+                    .first()
+                assertNotEquals(0, onlineAlternative.alternatives.size)
+            }
+        }
+    }
+
+    @Test
+    fun deviateFromOnlinePrimaryRouteWithoutInternet() = sdkTest {
+        val newRouteOnlineRouteRequestHandler = MockDirectionsRequestHandler(
+            profile = DirectionsCriteria.PROFILE_DRIVING_TRAFFIC,
+            jsonResponse = readRawFileText(context, R.raw.ev_routes_berlin_reroute),
+            expectedCoordinates = listOf(
+                Point.fromLngLat(13.36742058325467, 52.49745756017697),
+                evRouteInBerlin().coordinatesList().last()
+            )
+        )
+        mockWebServerRule.requestHandlers.add(newRouteOnlineRouteRequestHandler)
+
+        withMapboxNavigationAndOfflineTilesForRegion(
+            OfflineRegions.Berlin
+        ) { navigation ->
+            navigation.startTripSession()
+            val requestResult = navigation.requestRoutes(evRouteInBerlin())
+                .getSuccessfulResultOrThrowException()
+            assertEquals(RouterOrigin.Offboard, requestResult.routerOrigin)
+            navigation.setNavigationRoutesAsync(requestResult.routes)
+            withoutInternet {
+                stayOnPosition( // off route position
+                    longitude = 13.36742058325467,
+                    latitude = 52.49745756017697
+                ) {
+                    val newRoutes = navigation.routesUpdates()
+                        .first { it.reason == RoutesExtra.ROUTES_UPDATE_REASON_REROUTE }
+                    assertEquals(RouterOrigin.Onboard, newRoutes.navigationRoutes.first().origin)
+                }
+            }
+            stayOnPosition( // origin position after reroute
+                longitude = 13.36742058325467,
+                latitude = 52.49745756017697
+            ) {
+                // TODO: NN doesn't respect base url and always uses api.mapbox.com
+                val firstOnlineAlternative = navigation.alternativesUpdates()
+                    .filterIsInstance<NavigationRouteAlternativesResult.OnRouteAlternatives>()
+                    .filter { it.routerOrigin == RouterOrigin.Offboard }
+                    .first()
```

Comment:
How do you guarantee this alternative is for `reroute` route, not the original one? The callback is sticky, as far as I remember, and it may fire with an old value.

Reply:
not actual any more